### ISSUE TITLE
List volumes endpoint errors if any volume disappears while presenting

### ISSUE THREAD

**Issue by vito** [concourse/concourse#1473]
We currently make 3 DB calls per volume, making this endpoint incredibly slow and error prone:

https://github.com/concourse/atc/blob/master/api/present/volume.go#L9-L23

**Comment by jtarchie** [comment]
We are going to make this a single query, not an N+1.

**Comment by jtarchie** [comment]
We saw that when any of volumes errored (ie: something got deleted) the entire endpoint failed. We've fixed it up so that it only returns volumes that haven't errored, logs the errored volumes, and get's you ice cream (_not really_).

We also took a crack, eh, at unifying the query for the endpoint, so that we wouldn't run into this issue. We have a single query that can return the necessary information. The inclusion of the query would change quite a bit of modeling outside this endpoint -- mainly `db/volume.go`.

If we want to move on with a remodeling, let's discuss and schedule some work.

And my pair will copy+pasta a nice SQL statement.
